android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BluetoothSite.java

```diff
@@ -193,9 +198,22 @@ public void onServicesDiscovered(BluetoothGatt gatt, int status) {
     }
 
     characteristic = gatt.getService(SERVICE_UUID).getCharacteristic(CHARACTERISTIC_WEBPAGE_UUID);
+    checkWebpageGzipped();
+    
     gatt.readCharacteristic(characteristic);
   }
 
+  private void checkWebpageGzipped() {
```

Comment:
See also GZIP_MAGIC here: http://stackoverflow.com/questions/4818468/how-to-check-if-inputstream-is-gzipped